.babelrc

```diff
@@ -6,5 +6,9 @@
       },
       "exclude": ["transform-regenerator"]
     }]
+  ],
+  "plugins": [
+    "transform-object-rest-spread",
```

Comment:
i think this might not be needed if you change the node preset above from 4 to 18?

Reply:
It seems to be an error.

![image](https://github.com/user-attachments/assets/00a7e244-6942-4dec-9883-9d546cbf8ea8)

```
> mocha 'test/bin/**/*.js' 'test/lib/**/*.js' --recursive

/home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/transformation/file/index.js:558
      throw err;
      ^

SyntaxError: /home/study/workspace/eslint-find-rules/test/lib/rule-finder.js: Unexpected token (221:6)
  219 |         '@global': true
  220 |       },
> 221 |       ...mock
      |       ^
  222 |     })
  223 |   : (specifiedFileRelative, options) => {
  224 |       const config = specifiedFileRelative 
    at Parser.pp$5.raise (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:4454:13)
    at Parser.pp.unexpected (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:1761:8)
    at Parser.pp$3.parseIdentifier (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:4332:10)
    at Parser.pp$3.parsePropertyName (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:4156:96)
    at Parser.pp$3.parseObj (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:4045:12)
    at Parser.pp$3.parseExprAtom (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3719:19)
    at Parser.pp$3.parseExprSubscripts (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3494:19)
    at Parser.pp$3.parseMaybeUnary (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3474:19)
    at Parser.pp$3.parseExprOps (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3404:19)
    at Parser.pp$3.parseMaybeConditional (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3381:19)
    at Parser.pp$3.parseMaybeAssign (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3344:19)
    at Parser.pp$3.parseExprListItem (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:4312:16)
    at Parser.pp$3.parseCallExpressionArguments (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3573:20)
    at Parser.pp$3.parseSubscripts (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3533:31)
    at Parser.pp$3.parseExprSubscripts (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3504:15)
    at Parser.pp$3.parseMaybeUnary (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3474:19)
    at Parser.pp$3.parseExprOps (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3404:19)
    at Parser.pp$3.parseMaybeConditional (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3381:19)
    at Parser.pp$3.parseMaybeAssign (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3344:19)
    at Parser.pp$3.parseConditional (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3391:28)
    at Parser.pp$3.parseMaybeConditional (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3384:15)
    at Parser.pp$3.parseMaybeAssign (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3344:19)
    at Parser.pp$1.parseVar (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:2340:24)
    at Parser.pp$1.parseVarStatement (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:2169:8)
    at Parser.pp$1.parseStatement (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:1861:19)
    at Parser.pp$1.parseBlockBody (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:2268:21)
    at Parser.pp$1.parseTopLevel (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:1778:8)
    at Parser.parse (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:1673:17)
    at parse (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:7305:37)
    at File.parse (/home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/transformation/file/index.js:485:15)
    at File.parseCode (/home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/transformation/file/index.js:570:20)
    at /home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/transformation/pipeline.js:49:12
    at File.wrap (/home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/transformation/file/index.js:532:16)
    at Pipeline.transform (/home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/transformation/pipeline.js:47:17)
    at Object.transformFileSync (/home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/api/node.js:152:10)
    at compile (/home/study/workspace/eslint-find-rules/node_modules/babel-register/lib/node.js:118:20)
    at loader (/home/study/workspace/eslint-find-rules/node_modules/babel-register/lib/node.js:144:14)
    at Object.require.extensions.<computed> [as .js] (/home/study/workspace/eslint-find-rules/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (node:internal/modules/cjs/loader:1203:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1019:12)
    at Module.require (node:internal/modules/cjs/loader:1231:19)
    at require (node:internal/modules/helpers:177:18)
    at /home/study/workspace/eslint-find-rules/node_modules/mocha/lib/mocha.js:231:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/home/study/workspace/eslint-find-rules/node_modules/mocha/lib/mocha.js:228:14)
    at Mocha.run (/home/study/workspace/eslint-find-rules/node_modules/mocha/lib/mocha.js:514:10)
    at Object.<anonymous> (/home/study/workspace/eslint-find-rules/node_modules/mocha/bin/_mocha:480:18)
    at Module._compile (node:internal/modules/cjs/loader:1364:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1422:10)
    at Module.load (node:internal/modules/cjs/loader:1203:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1019:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:128:12)
    at node:internal/main/run_main_module:28:49 {
  pos: 5680,
  loc: Position { line: 221, column: 6 },
  _babel: true,
  codeFrame: "\x1B[0m \x1B[90m 219 | \x1B[39m        \x1B[32m'@global'\x1B[39m\x1B[33m:\x1B[39m \x1B[36mtrue\x1B[39m\n" +
    ' \x1B[90m 220 | \x1B[39m      }\x1B[33m,\x1B[39m\n' +
    '\x1B[31m\x1B[1m>\x1B[22m\x1B[39m\x1B[90m 221 | \x1B[39m      \x1B[33m...\x1B[39mmock\n' +
    ' \x1B[90m     | \x1B[39m      \x1B[31m\x1B[1m^\x1B[22m\x1B[39m\n' +
    ' \x1B[90m 222 | \x1B[39m    })\n' +
    ' \x1B[90m 223 | \x1B[39m  \x1B[33m:\x1B[39m (specifiedFileRelative\x1B[33m,\x1B[39m options) \x1B[33m=>\x1B[39m {\n' +
    ' \x1B[90m 224 | \x1B[39m      \x1B[36mconst\x1B[39m config \x1B[33m=\x1B[39m specifiedFileRelative \x1B[0m'
}

Node.js v18.20.2
```

Reply:
bizarre… what if we set it to 20?

Reply:
```
/home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/transformation/file/index.js:558
      throw err;
      ^

SyntaxError: /home/study/workspace/eslint-find-rules/test/lib/rule-finder.js: Unexpected token (185:6)
  183 |         '@global': true
  184 |       },
> 185 |       ...mock
      |       ^
  186 |     })
  187 |   : (specifiedFileRelative, options) => {
  188 |       const config = specifiedFileRelative 
    at Parser.pp$5.raise (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:4454:13)
    at Parser.pp.unexpected (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:1761:8)
    at Parser.pp$3.parseIdentifier (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:4332:10)
    at Parser.pp$3.parsePropertyName (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:4156:96)
    at Parser.pp$3.parseObj (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:4045:12)
    at Parser.pp$3.parseExprAtom (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3719:19)
    at Parser.pp$3.parseExprSubscripts (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3494:19)
    at Parser.pp$3.parseMaybeUnary (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3474:19)
    at Parser.pp$3.parseExprOps (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3404:19)
    at Parser.pp$3.parseMaybeConditional (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3381:19)
    at Parser.pp$3.parseMaybeAssign (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3344:19)
    at Parser.pp$3.parseExprListItem (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:4312:16)
    at Parser.pp$3.parseCallExpressionArguments (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3573:20)
    at Parser.pp$3.parseSubscripts (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3533:31)
    at Parser.pp$3.parseExprSubscripts (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3504:15)
    at Parser.pp$3.parseMaybeUnary (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3474:19)
    at Parser.pp$3.parseExprOps (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3404:19)
    at Parser.pp$3.parseMaybeConditional (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3381:19)
    at Parser.pp$3.parseMaybeAssign (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3344:19)
    at Parser.pp$3.parseConditional (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3391:28)
    at Parser.pp$3.parseMaybeConditional (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3384:15)
    at Parser.pp$3.parseMaybeAssign (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:3344:19)
    at Parser.pp$1.parseVar (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:2340:24)
    at Parser.pp$1.parseVarStatement (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:2169:8)
    at Parser.pp$1.parseStatement (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:1861:19)
    at Parser.pp$1.parseBlockBody (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:2268:21)
    at Parser.pp$1.parseTopLevel (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:1778:8)
    at Parser.parse (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:1673:17)
    at parse (/home/study/workspace/eslint-find-rules/node_modules/babylon/lib/index.js:7305:37)
    at File.parse (/home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/transformation/file/index.js:485:15)
    at File.parseCode (/home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/transformation/file/index.js:570:20)
    at /home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/transformation/pipeline.js:49:12
    at File.wrap (/home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/transformation/file/index.js:532:16)
    at Pipeline.transform (/home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/transformation/pipeline.js:47:17)
    at Object.transformFileSync (/home/study/workspace/eslint-find-rules/node_modules/babel-core/lib/api/node.js:152:10)
    at compile (/home/study/workspace/eslint-find-rules/node_modules/babel-register/lib/node.js:118:20)
    at loader (/home/study/workspace/eslint-find-rules/node_modules/babel-register/lib/node.js:144:14)
    at Object.require.extensions.<computed> [as .js] (/home/study/workspace/eslint-find-rules/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (node:internal/modules/cjs/loader:1288:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1104:12)
    at Module.require (node:internal/modules/cjs/loader:1311:19)
    at require (node:internal/modules/helpers:179:18)
    at /home/study/workspace/eslint-find-rules/node_modules/mocha/lib/mocha.js:231:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/home/study/workspace/eslint-find-rules/node_modules/mocha/lib/mocha.js:228:14)
    at Mocha.run (/home/study/workspace/eslint-find-rules/node_modules/mocha/lib/mocha.js:514:10)
    at Object.<anonymous> (/home/study/workspace/eslint-find-rules/node_modules/mocha/bin/_mocha:480:18)
    at Module._compile (node:internal/modules/cjs/loader:1469:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
    at Module.load (node:internal/modules/cjs/loader:1288:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1104:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12)
    at node:internal/main/run_main_module:28:49 {
  pos: 4660,
  loc: Position { line: 185, column: 6 },
  _babel: true,
  codeFrame: "\x1B[0m \x1B[90m 183 | \x1B[39m        \x1B[32m'@global'\x1B[39m\x1B[33m:\x1B[39m \x1B[36mtrue\x1B[39m\n" +
    ' \x1B[90m 184 | \x1B[39m      }\x1B[33m,\x1B[39m\n' +
    '\x1B[31m\x1B[1m>\x1B[22m\x1B[39m\x1B[90m 185 | \x1B[39m      \x1B[33m...\x1B[39mmock\n' +
    ' \x1B[90m     | \x1B[39m      \x1B[31m\x1B[1m^\x1B[22m\x1B[39m\n' +
    ' \x1B[90m 186 | \x1B[39m    })\n' +
    ' \x1B[90m 187 | \x1B[39m  \x1B[33m:\x1B[39m (specifiedFileRelative\x1B[33m,\x1B[39m options) \x1B[33m=>\x1B[39m {\n' +
    ' \x1B[90m 188 | \x1B[39m      \x1B[36mconst\x1B[39m config \x1B[33m=\x1B[39m specifiedFileRelative \x1B[0m'
}

Node.js v20.18.0
```

Reply:
ok, keep it at 18, but preset env should be handling this.

I’ll change exports myself, after landing this.

Reply:
i changed targets.node to 18